staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -189,6 +193,7 @@ type Config struct {
 	LivezChecks []healthz.HealthChecker
 	// The default set of readyz-only checks. There might be more added via AddReadyzChecks dynamically.
 	ReadyzChecks []healthz.HealthChecker
+	FlagSets     []*cliflag.NamedFlagSets
```

Comment:
It's pretty strange to give access to the actual flagsets here, I *think* this could allow things with a handle to the options to mutate flags, which we *really* don't want

Thinking of this instead as an input *only* to the flagz handler, this could be an interface (defined in flagz) with a function that returns `[]flagz.Flag` and is just handled directly to `flagz.Install` (if non-nil)

Then an adapter that wraps NamedFlagSets and implements `GetFlagz() []flagz.Flag` (which calls ConvertNamedFlagSetToFlags) lets us encapsulate the NamedFlagSets without exposing them to mutation or to anything other than use by flagz.

That also has the benefit of letting us shift the conversion from NamedFlagSets → []flagz.Flag later to first hit of `/flagz` instead of at install time, if we want to

Reply:
Thanks for this suggestion, I do see the benefit of encapsulating NamedFlagSets behind an interface and making it read-only. 

I have tried to follow what you described and now have a flagz.Reader interface that has  a `GetFlagz() []Flag` method. Also have defined a `NamedFlagSetsReader` struct in flagz that implements this interface.